drivers/nhs-nut.c

```diff
@@ -0,0 +1,2070 @@
+                strcpy(data.upsdesc,"LASER ONLINE 12000VA");
+                data.VA = 12000;
+        break;
+
+        case 103:
+                data.upscode = 103;
+                strcpy(data.upsdesc,"LASER ONLINE 8000VA");
+                data.VA = 8000;
+        break;
+
+        case 104:
+                data.upscode = 104;
+                strcpy(data.upsdesc,"PDV SENOIDAL ISOLADOR 1000VA");
+                data.VA = 1000;
+        break;
+
+        case 105:
+                data.upscode = 105;
+                strcpy(data.upsdesc,"MINI SENOIDAL 500VA");
+                data.VA = 500;
+        break;
+
+        case 106:
+                data.upscode = 106;
+                strcpy(data.upsdesc,"LASER SENOIDAL 5000VA GII");
+                data.VA = 5000;
+        break;
+
+        case 107:
+                data.upscode = 107;
+                strcpy(data.upsdesc,"COMPACT PLUS SENOIDAL 1000VA");
+                data.VA = 1000;
+        break;
+
+        case 108:
+                data.upscode = 108;
+                strcpy(data.upsdesc,"QUAD_COM 80A");
+                data.VA = 0;
+        break;
+
+        case 109:
+                data.upscode = 109;
+                strcpy(data.upsdesc,"LASER ONLINE 5000VA");
+                data.VA = 5000;
+        break;
+
+        case 113:
+                data.upscode = 113;
+                strcpy(data.upsdesc,"PDV SENOIDAL ISOLADOR 700VA");
+                data.VA = 700;
+        break;
+
+        default:
+                data.upscode = -1;
+                strcpy(data.upsdesc,"NHS UNKNOWN");
+                data.VA = 0;
+        break;
+    }
+    return data;
+}
+
+unsigned int get_va(int equipment) {
+    upsinfo ups;
+    char * va = getval("va");
+    ups = getupsinfo(equipment);
+    if (ups.VA > 0)
+        return ups.VA;
+    else {
+        if (va)
+            return atoi(va);
+        else
+            fatalx(EXIT_FAILURE,"Please set VA (Volt Ampere) nominal capacity value to your equipment in ups.conf.");
+    }
+}
+
+float get_pf() {
+    char * pf = getval("pf");
+    if (pf)
+        return atof(pf);
+    else
+        return DEFAULTPF;
+}
+
+unsigned int get_ah() {
+    char * ah = getval("ah");
+    if (ah)
+        return (unsigned int)atoi(ah);
+    else
+        fatalx(EXIT_FAILURE,"Please set AH (Ampere Hour) value to your battery's equipment in ups.conf.");
+}
+
+float get_vin_perc(char * var) {
+    char * perc = getval(var);
+    if (perc)
+        return atof(perc);
+    else
+        return DEFAULLTPERC;
+}
+
+void upsdrv_initinfo(void) {
+    char * b = getval("baud");
+    upsdebugx(1,"Port is %s and baud_rate is %s",device_path,b);
+    baudrate = DEFAULTBAUD;
+    upsdebugx(1,"Start initinfo()");
+    if (b)
+        baudrate = atoi(b);
+    if (device_path) {
+        if (strcasecmp(device_path,"auto") == 0)
+            strcpy(porta,DEFAULTPORT);
+        else
+            strcpy(porta,device_path);
+        serial_fd = openfd(porta,baudrate);
```

Comment:
`upsdrv_initups()` should take care of hardware communications init (maybe fail with `fatalx()` if `port` is not accessible).

The `upsdrv_initinfo()` method is called later to detect the device details (maybe fail with `fatalx()` if no compatible device was found) and set write-once values like device model etc. and populate the list of supported commands and/or settable variables, if any.

Please do check `docs/new-drivers.txt` for general driver code structure overview.